src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -38,139 +43,1174 @@ const ActorInit En_Ta_InitVars = {
+    } else {
+        if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+            SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, this->unk_2E4);
+            this->unk_2E2 = Math_Rand_ZeroFloat(100.0f) + 100.0f;
+        }
+
+        if (this->skelAnime.animCurrentFrame < 96.0f && this->skelAnime.animCurrentFrame >= 53.0f) {
+            this->unk_2B4 = 1;
+        } else {
+            this->unk_2B4 = 2;
+        }
+        this->unk_2E0 |= 8;
+    }
+    this->unk_2E0 |= 4;
+}
+
+void func_80B16938(EnTa* this) {
+    if (!(this->unk_2E0 & 0x10)) {
+        if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+            this->unk_2E0 |= 0x10;
+        }
+        this->unk_2E0 |= 8;
+    }
+}
+
+void EnTa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTa* this = THIS;
+    s32 pad;
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->unk_260(this);
+    this->unk_25C(this, globalCtx);
+
+    if (!(this->unk_2E0 & 4)) {
+        this->unk_2B0(this);
+    }
+
+    if (this->unk_2E0 & 1) {
+        func_80038290(globalCtx, &this->actor, &this->unk_2D4, &this->unk_2DA, this->actor.posRot2.pos);
+    } else {
+        Math_SmoothScaleMaxMinS(&this->unk_2D4.x, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2D4.y, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2DA.x, 0, 6, 6200, 100);
+        Math_SmoothScaleMaxMinS(&this->unk_2DA.y, 0, 6, 6200, 100);
+    }
+
+    this->unk_2E0 &= ~0x5;
+
+    if (this->unk_2CC > 0) {
+        this->unk_2CC--;
+    }
+}
+
+s32 EnTa_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+    EnTa* this = THIS;
+
+    switch (limbIndex) {
+        case 8:
+            rot->x += this->unk_2DA.y;
+            rot->y -= this->unk_2DA.x;
+            break;
+        case 15:
+            rot->x += this->unk_2D4.y;
+            rot->z += this->unk_2D4.x;
+            break;
+    }
+
+    if (this->unk_2E0 & 0x8) {
+        this->unk_2E0 &= ~0x8;
+    } else if ((limbIndex == 8) || (limbIndex == 10) || (limbIndex == 13)) {
+        s32 limbIdx50 = limbIndex * 50;
+        rot->y += Math_Sins(globalCtx->state.frames * (limbIdx50 + 0x814)) * 200.0f;
+        rot->z += Math_Coss(globalCtx->state.frames * (limbIdx50 + 0x940)) * 200.0f;
+    }
+
+    return 0;
+}
+
+void EnTa_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    if (limbIndex == 15) {
+        Matrix_MultVec3f(&D_80B16E7C, &((Actor*)thisx)->posRot2.pos);
+    }
+}
+
+void EnTa_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnTa* this = THIS;
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_ta.c", 2381);
+
+    func_800943C8(globalCtx->state.gfxCtx);
+    if (thisx) {};
+
```

Comment:
does passing thisx as the last call of skelanime draw allow you to remove this? if not what about using globalCtx2?

Reply:
Turns out the if (thisx) wasn't needed at all...lol